packages/playwright-test/src/worker.ts

```diff
@@ -126,12 +127,24 @@ function initConsoleParameters(initParams: WorkerInitParams) {
   // Make sure the output supports colors.
   process.stdout.isTTY = true;
   process.stderr.isTTY = true;
-  if (initParams.stdoutDimension.rows)
-    process.stdout.rows = initParams.stdoutDimension.rows;
-  if (initParams.stdoutDimension.columns)
-    process.stdout.columns = initParams.stdoutDimension.columns;
-  if (initParams.stderrDimension.rows)
-    process.stderr.rows = initParams.stderrDimension.rows;
-  if (initParams.stderrDimension.columns)
-    process.stderr.columns = initParams.stderrDimension.columns;
+  if (initParams.stdoutParams.rows)
+    process.stdout.rows = initParams.stdoutParams.rows;
+  if (initParams.stdoutParams.columns)
+    process.stdout.columns = initParams.stdoutParams.columns;
+  setColorDepthMethods(process.stdout, initParams.stdoutParams.colorDepth);
+  if (initParams.stderrParams.rows)
+    process.stderr.rows = initParams.stderrParams.rows;
+  if (initParams.stderrParams.columns)
+    process.stderr.columns = initParams.stderrParams.columns;
+  setColorDepthMethods(process.stderr, initParams.stderrParams.colorDepth);
+}
+
+function setColorDepthMethods(stream: WriteStream, depth: number) {
```

Comment:
Let's reduce duplication by generalizing this method to `setTTYParams(stream, params)`